### ISSUE TITLE
Improve code with splint

### ISSUE THREAD

**Issue by Lokaltog** [Lokaltog/candybar#32]
Tried running some of our source code through splint, and it spews out a ton of warnings. Some of these warnings should probably be looked into (buffer overflow issues, possible null pointers, etc). Here's a command that can be run from the project root:

```
splint [file.c] -I. -I `pkg-config --cflags gtk+-3.0 webkitgtk-3.0`
```

[Output for src/wkline.c](https://gist.github.com/Lokaltog/9397502)
[Output for src/wkline/widgets.c](https://gist.github.com/Lokaltog/9397516)
##

**Comment by nHurD** [comment]
[Valgrind](http://www.valgrind.org) is another useful tool for checking memory leaks and the like. It also may be a bit disheartening to run at times...

**Comment by Lokaltog** [comment]
Yep, I've tried to use it a couple of times on wkline. It's pretty annoying to run with webkit, it spits out a ton of warnings and most of them stem from somewhere in the webkit engine.
